astropy/cosmology/core.py

```diff
@@ -93,8 +96,12 @@ class Cosmology(metaclass=abc.ABCMeta):
     write = UnifiedReadWriteMethod(CosmologyWrite)
 
     # Parameters
-    __parameters__: tuple[str, ...] = ()
-    __all_parameters__: tuple[str, ...] = ()
+    parameters = ParameterDescriptor()
```

Comment:
For the docstrings, one solution for now might be to just give explicit docstrings here.

Reply:
Done!